tests/stdlib/tlists.nim

```diff
@@ -95,25 +95,51 @@ block toDoublyLinkedList:
   doAssert [1, 2, 3].toDoublyLinkedList.toSeq == [1, 2, 3]
 
 block add:
-  block:
-    var
-      l0 = initSinglyLinkedList[int]()
-      l1 = [1].toSinglyLinkedList
-      l2 = [2, 3].toSinglyLinkedList
-      l3 = [4, 5, 6].toSinglyLinkedList
-    l0.add(l3)
-    l1.add(l3)
-    l2.add(l3)
-    doAssert l0.toSeq == [4, 5, 6]
-    doAssert l1.toSeq == [1, 4, 5, 6]
-    doAssert l2.toSeq == [2, 3, 4, 5, 6]
-  block:
-    var
-      l0 = initSinglyLinkedList[int]()
-      l1 = [1].toSinglyLinkedList
-      l2 = [2, 3].toSinglyLinkedList
-      l3 = [4, 5, 6].toSinglyLinkedList
-    l3.add(l0)
-    l2.add(l1)
-    doAssert l3.toSeq == [4, 5, 6]
-    doAssert l2.toSeq == [2, 3, 1]
+  block: # SinglyLinkedList
+    block:
+      var
+        l0 = initSinglyLinkedList[int]()
+        l1 = [1].toSinglyLinkedList
+        l2 = [2, 3].toSinglyLinkedList
+        l3 = [4, 5, 6].toSinglyLinkedList
+      l0.add(l3.copy)
+      l1.add(l3.copy)
+      l2.add(l3)
+      doAssert l0.toSeq == [4, 5, 6]
+      doAssert l1.toSeq == [1, 4, 5, 6]
+      doAssert l2.toSeq == [2, 3, 4, 5, 6]
+      doAssert l3.toSeq == []
```

Comment:
add:
```nim
l2.add(l3) # re-adding l3 that was destroyed is now a noop
doAssert l3.toSeq == []
doAssert l2.toSeq == [2, 3, 4, 5, 6]
```

Reply:
Sure, but that seems pretty self-explanatory from `l3.toSeq == []` and the `l3.add(l0)` example in the other block.

Reply:
but the test file currently doesn't even exercise the case where you're appending an empty list; test files should test edge cases to prevent regressions from future implementation changes (or seemingly unrelated changes).

so actually I'd add that snippet, as well as a snippet that calls `a.add b` where `a` is empty.

Reply:
Empty sets on both the left- and right-hand sides are tested (`l0.add(l3)` in the first sub-block, and vice versa in the second sub-block).

Reply:
ok!